File: hw/camacho/u2/Hw29ZooApp/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Cat.java
package ec.edu.espe.zooapp.model;

/**
 *
 * @author mateo
 */
public class Cat extends Animal {
    private String furColor;
    private boolean isIndoor;

    public Cat(String name, int age, String species, String furColor, boolean isIndoor) {
        super(name, age, species);
        this.furColor = furColor;
        this.isIndoor = isIndoor;
    }

    public String getFurColor() {
        return furColor;
    }

    public void setFurColor(String furColor) {
        this.furColor = furColor;
    }

    public boolean isIndoor() {
        return isIndoor;
    }

    public void setIndoor(boolean isIndoor) {
        this.isIndoor = isIndoor;
    }
}

File: ws/camacho/u3/JavaScriptPet/model/Cat.js
import Animal from './Animal.js';

class Cat {
    constructor(name, weight) {
        return new Animal({
            name,
            weight,
            type: 'cat',
            foodAmount: Cat.calculateFood(weight)
        });
    }

    static calculateFood(weight) {
        return weight * 0.03; 
    }
}

export default Cat;